reposerver/src/main/scala/com/advancedtelematic/tuf/reposerver/http/RepoResource.scala

```diff
@@ -229,15 +229,19 @@ class RepoResource(keyserverClient: KeyserverClient, namespaceValidation: Namesp
     (pathPrefix("user_repo") & namespaceValidation.extractor) { namespace =>
       (post & pathEnd) {
         val repoId = RepoId.generate()
-        createRepo(namespace, repoId)
+        keyTypeParameter { keyType =>
+          createRepo(namespace, repoId, keyType.getOrElse(defaultKeyType))
+        }
       } ~
       UserRepoId(namespace) { repoId =>
         modifyRepoRoutes(repoId)
       }
     } ~
       (pathPrefix("repo" / RepoId.Path) & namespaceValidation.extractor) { (repoId, namespace) =>
         (pathEnd & post) {
```

Comment:
on a post request the body of the request should provide the parameters, not the url. See https://github.com/advancedtelematic/ota-tuf/blob/master/keyserver/src/main/scala/com/advancedtelematic/tuf/keyserver/http/RootRoleResource.scala#L119

Reply:
Can an entity be optional?

Reply:
That workaround with multiple routes doesn't look that bad...